sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -499,21 +509,26 @@ public void close() {
                 sink.error(throwable);
             })
             .switchIfEmpty(Mono.error(new AmqpException(true, "No response received from management channel.",
-                channel.getErrorContext())));
+                channel.getErrorContext())))
+            .onErrorMap(this::mapError);
     }
 
     private Mono<Void> isAuthorized(String operation) {
         return tokenManager.getAuthorizationResults()
             .next()
             .handle((response, sink) -> {
                 if (response != AmqpResponseCode.ACCEPTED && response != AmqpResponseCode.OK) {
-                    sink.error(new AmqpException(false, String.format(
-                        "User does not have authorization to perform operation [%s] on entity [%s]. Response: [%s]",
-                        operation, entityPath, response), getErrorContext()));
+                    final String message = String.format("User does not have authorization to perform operation "
+                        + "[%s] on entity [%s]. Response: [%s]", operation, entityPath, response);
+                    final Throwable exc = new AmqpException(false, AmqpErrorCondition.UNAUTHORIZED_ACCESS,
+                        message, getErrorContext());
+                    sink.error(new ServiceBusException(exc, ServiceBusErrorSource.MANAGEMENT));
                 } else {
                     sink.complete();
                 }
-            });
+            })
+            .onErrorMap(this::mapError)
+            .then();
```

Comment:
I don't think `then()` is needed here.

Reply:
You were right, it needed to be hoisted further up!